src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1550,17 +1550,11 @@ codet java_bytecode_convert_methodt::convert_instructions(
         symbol.type=arg0.type();
         symbol.value.make_nil();
         symbol.mode=ID_java;
-
         assign_parameter_names(
           to_code_type(symbol.type),
           symbol.name,
           symbol_table);
 
-        // The string refinement module may provide a definition for this
-        // function.
-        symbol.value=string_preprocess.code_for_function(
```

Comment:
The method that is added below (`replace_string_methods`) does the same thing. But instead of doing it in the middle of `convert_instructions` it does so at the end of the bytecode conversion by going through the symbol table.
The potential change in semantics is that we now convert methods that were forgotten before because we were not getting in this particular branch of `convert_instructions`, but this should be an improvement.